12-target-details.tex

```diff
@@ -5,6 +5,144 @@ \section{Javascript}
 \label{target-javascript}
 \state{NoContent}
 
+\subsection{Getting started with Javascript}
+\label{target-javascript-getting-started}
+
+Developing Javascript applications is really powerful with Haxe. Let's see our first sample.
+This is a very simple example showing the toolchain. 
+
+Create a new folder, save this class as \ic{Main.hx}.
+
+\begin{lstlisting}
+import js.Lib;
+import js.Browser;
+class Test {
+    static function main() {
+        var button = Browser.document.createButtonElement();
+        button.textContent = "Click me!";
+        button.onclick = function(event) {
+            Lib.alert("Haxe is great");
+        }
+        Browser.document.body.appendChild(button);
+    }
+}
+\end{lstlisting}
+
+To compile, run this from the command line:
+
+\begin{lstlisting}
+haxe -js main-javascript.js -main Main -D js-flatten -dce full
+\end{lstlisting}
+
+.. or create and run (double-click) a file called \ic{compile.hxml}. In this example the hxml-file should be in the same directory as the example class.
+
+\begin{lstlisting}
+-js main-javascript.js
+-main Main
+-D js-flatten
+-dce full
+\end{lstlisting}
+
+The output will be a main-javascript.js, which creates and adds clickable button to the document body.
+
+\paragraph{Run the Javascript}
+
+To display the output in a browser, create a HTML-document called \ic{index.html} and open it.
+
+\begin{lstlisting}
+<!DOCTYPE html>
+<html>
+	<body>
+		<script src="main-javascript.js">
+	</body>
+</html>
+\end{lstlisting}
+
+\paragraph{More information}
+
+\begin{itemize}
+	\item \href{http://api.haxe.org/js/}{Haxe Javascript API docs}
+	\item \href{https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference}{MDN JavaScript Reference}
+\end{itemize}
+
+\subsection{Using external Javascript libraries}
+\label{target-javascript-external-libraries}
+
+The Haxe Standard Library comes with many externs for the Javascript target. They allow accessing the native APIs in a type-safe manner.
+The \tref{externs mechanism}{lf-externs} assumes that the defined types exist at runtime, but assumes nothing about how and where those types are defined. 
+To clarify, in most cases we have to add a \ic{<script>}-tag that links to the external library manually to the HTML-document.
+
+
+An example of an extern class is the \href{https://github.com/HaxeFoundation/haxe/blob/development/std/js/JQuery.hx#L83}{jQuery class} of the Standard Library. 
+To illustrate here is a part of this extern class:
```

Comment:
"To illustrate, ..." or "Here is .....to ilustrate" or "For the purpose of illustration" or some such.